teps/0007-conditions-beta.md

```diff
@@ -0,0 +1,609 @@
+
+Here's how a user can specify a `Guard` that executes on that `Result`: 
+
+```yaml
+apiVersion: tekton.dev/v1alpha1
+kind: Task
+metadata:
+  name: file-exists
+spec:
+  params:
+    - name: path
+  workspaces:
+    - name: source
+  results:
+    - name: exists
+      description: boolean indicating whether the file exists
+  steps:
+  - name: check-file-exists
+    image: alpine
+    script: |
+      if [ -f $(workspaces.source.path)/$(params.path) ]; then
+        echo true | tee /tekton/results/exists
+      else
+        echo false | tee /tekton/results/exists
+      fi
+---
+api: tekton.dev/v1alpha1
+kind: Pipeline
+metadata: 
+    name: generate-file
+spec:
+  workspaces:
+    - name: source-repo
+  params:
+    - name: path
+      default: 'README.md'
+  tasks: 
+    - name: create-file
+      taskRef: 
+        name: create-readme-file
+      workspaces:
+        - name: source
+          workspace: source-repo
+    - name: file-exists
+      taskRef: 
+        name: file-exists
+      workspaces:
+        - name: source
+          workspace: source-repo
+    - name: echo-file-exists 
+      when:
+        - name: check-file-exists
+          input: '$(tasks.file-exists.results.exists)'
+          operator: In
+          values: ['true']
+      taskRef:
+        name: echo-file-exists
+```
+
+Other examples of `Guards` using `Match Expressions` are:
+
+```yaml
+name: branch-is-main
+input: $(params.branch)
+operator: In
+values: [‘main’]
+---
+name: branch-not-main
+input: $(params.branch)
+operator: NotIn
+values: [‘main’]
+---
+name: always-false
+input: ‘false’
+operator: In
+values: [‘’]
+```
+
+We can explore adding more `Operators` later if needed, such as `IsTrue`, `IsFalse`, `IsEmpty` and `IsNotEmpty`.
+
+### Skipping
+
+Provide more flexibility when a `Guard` executes as `False` by adding a field, `continueAfterSkip`, used to specify whether execute the `Tasks` that are ordering-dependent on the skipped guarded `Task`. The `continueAfterSkip` field defaults to `false`/`no` and users can set it to `true`/`yes` (case insensitive) to allow for execution of the rest of the branch. 
```

Comment:
i.e. `continueAfterSkip` only supported with Guarded tasks? `continueAfterSkip` can not be specified without any Guards, if specified, result in validation failure?